src/components/SearchInput/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import ClassNames from 'classnames';
+
+import styles from '@/components/SearchInput/SearchInput.css';
+
+import IconSearch from '@/assets/images/icons/web/search-simple.svg';
+import IconClose from '@/assets/images/icons/web/close.svg';
+
+type Size = 'large' | 'medium';
+type State = '' | 'valid' | 'invalid';
+
+interface Props {
+  clearInputClick: () => void;
+  description?: string;
+  disabled?: boolean;
+  label?: string;
+  message?: string;
+  onChange: () => void;
+  placeholder?: string;
+  state?: State;
+  size: Size;
+  value?: string;
+  variablesClassName?: string;
+}
+
+const SearchInput: React.FC<Props> = props => {
+  const {
+    clearInputClick,
+    description,
+    disabled,
+    label,
+    onChange,
+    message,
+    placeholder,
+    size,
+    state,
+    value,
+    variablesClassName,
+    ...searchInputProps
+  } = props;
+  const sizeClass = `search--${size}`;
+  const statusClass = state ? `search--${state}` : '';
+  const sizeInputClass = `search-input--${size}`;
+  const messageValidateClass = state ? `search-input--message-${state}` : '';
+
+  return (
+    <div className={ClassNames(variablesClassName)}>
+      {label ? <label className={ClassNames(styles['search-label'])}>{label}</label> : ''}
+      {description ? (
+        <span className={ClassNames(styles['search-description'])}>{description}</span>
+      ) : (
+        ''
+      )}
+      <div className={ClassNames(styles.search, styles[sizeClass], styles[statusClass])}>
+        <button className={ClassNames(styles['search-icon-button'])}>
+          <IconSearch height='15' width='15' title='Search Icon' />
+          <span className={ClassNames(styles['search-accessible-description'])}>
+            Search Icon on the left of the input text
+          </span>
+        </button>
+        <input
```

Comment:
Taking a look at google's search bar, I found the following that can improve a11y

- You could include an `aria-label="Search"`, this has to be used carefully, it could be either `Search` or have the same value as the `placeholder`. This aria-label could be included in case the `label` prop is missing. This will tell the user the context for this input
- I'd suggest adding an `onSearch/onSubmit` optional function too, users are used to navigating through the keyboard, we could add an `onKeyPress` event to the input, when we detect the key `enter` was pressed, we call the `onSearch` function. I'd like @kaiomagalhaes second opinion on this

Reply:
I think it would be great, we could have other ways to navigate the page

Reply:
since this will not be addressed now, should we include this as tech debt? @Rheniery @kaiomagalhaes